### ISSUE TITLE
Function implementation: manage and update pump state

### ISSUE THREAD

**Issue by lamara11** [amosproj/amos2024ss01-xcelerator-demo-app#65]
## User story

1. As a facility manager
2. I want to SDs to implement a function for suspicious pumps
3. So that I can identify if pumps might get faulty

## Acceptance criteria

- [ ] The function checks all pumps of a facility
- [ ] It sets a pump to "suspicious" if flow drops below 150
- [ ] It changes a pump from "suspicious" to "faulty" if flow drops below 150 for over 10 minutes
- [ ] If there is a rise in Stuffing Box Temperature and stable flow (deviation of 10), then set the pump to "suspicious"
- [ ] Add the attribute status in the backend and also add this to the interface so it can be used in the frontend

## Definition of done (DoD)

- [ ] Added only after week 5
- [ ] The same for all features
- [ ] Here goes the project specific part

## DoD general criteria

- [ ] Feature has been fully implemented
- [ ] Feature has been merged into the mainline
- [ ] All acceptance criteria were met
- [ ] Product owner approved features
- [ ] All tests are passing
- [ ] Developers agreed to release

**Comment by Sungonogi** [comment]
Details about the implementation:
Right now we use [ 'success', 'warning', 'critical' ] as our possible states because like this the cards look good. So to make things more readable you should consider naming the status similar too. And I think these 3 make sense even though the Siemens Api only gives us success/warning, because in our own mocked data created in #192 we might also have a critical one


![Image](https://github.com/amosproj/amos2024ss01-xcelerator-demo-app/assets/166606862/9001814f-b652-4e68-97f7-54ab9677d346)
